st2client/st2client/commands/action.py

```diff
@@ -523,7 +523,7 @@ def transform_object(value):
                     result[key] = value
             return result
 
-        def transform_array(value):
+        def transform_array(value, action_params={}):
```

Comment:
Please default it to `None` in the function signature and default it to empty dict inside a function otherwise you will be in for a surprise - https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html.

Reply:
This one still needs to be fixed^

Besides that, LGTM.

Reply:
I'm so sorry for overlooking the point to fix.
I'll do it right now and recheck others.